src/web/components/SharingPermission/BulkAddPermissions.tsx

```diff
@@ -42,70 +46,135 @@ export function BulkAddPermissions({
   };
 
+  const bulkAddPermissionsMessage = (
+    <>
+      <b>{getRecommendationMessageFromTypeNames(currentParticipantTypeNames)}</b>
+      <p>
+        Allow all current and future participants within a participation type to decrypt your UID2
+        tokens.
+      </p>
+    </>
+  );
+
+  const viewRecommendedParticipantsButton = (
+    <button
+      type='button'
+      className='text-button bulk-add-permissions-view-recommendations-button'
+      onClick={onToggleViewRecommendedParticipants}
+    >
+      {showRecommendedParticipants ? 'Hide Participants' : 'View Participants'}
+    </button>
+  );
+
+  const participantTypeCheckboxes = (
+    <div className='participant-type-checkbox-section'>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('publisherChecked')}
+      />
+      <span className='checkbox-label'>Publisher</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('advertiserChecked')}
+      />
+      <span className='checkbox-label'>Advertiser</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('DSPChecked')}
+      />
+      <span className='checkbox-label'>DSP</span>
+      <FormStyledCheckbox
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...register('dataProviderChecked')}
+      />
+      <span className='checkbox-label'>Data Provider</span>
+    </div>
+  );
+
+  const saveRecommendationsButton = (
+    <div className='bulk-add-permissions-footer'>
+      <button type='submit' className='primary-button bulk-add-permissions-submit-button'>
+        Save Recommendations
+      </button>
+    </div>
+  );
+
+  const recommendationContent = (
+    <div className='bulk-add-permissions'>
+      <div className='bulk-add-permissions-body'>
+        {bulkAddPermissionsMessage}
+        {participantTypeCheckboxes}
+        {viewRecommendedParticipantsButton}
+      </div>
+      {saveRecommendationsButton}
+    </div>
+  );
+
+  const collapsibleContent = (
+    <div className='bulk-add-permissions'>
+      <div className='bulk-add-permissions-body'>
+        {bulkAddPermissionsMessage}
+        {participantTypeCheckboxes}
+        {hasUncheckedDefaultParticipantTypeRecommendation(
+          currentParticipantTypeNames,
+          watchPublisherChecked,
+          watchAdvertiserChecked,
+          watchDSPChecked,
+          watchDataProviderChecked
+        ) && (
+          <div className='remove-recommended-type-warning'>
+            <Banner
+              type='warning'
+              message='If you remove the sharing permissions for a participant type, all sharing permissions of that type are removed, including future participants of that type.'
+            />
+          </div>
+        )}
+      </div>
+      {isDirty && saveRecommendationsButton}
```

Comment:
Do we need to share the button if no changes is made?

Reply:
The design shows a state in which the button isn't shown if no changes are made, unless I am misinterpreting something.

![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/46cd1cae-06d4-44e2-ae67-eb7d753dad30)

Reply:
Oh I see, I didn't find this design before

Reply:
It's a trade-off - on one hand, we save some vertical space if they haven't changed anything. On the other hand, this means the content below will all shift up and down as the user makes changes/reverts their changes. 🤷

I think wait and see if the reflow becomes annoying - if it is annoying, there might be a case for always having the button, but leaving it disabled if there are no changes.